provider/sidecar_test.go

```diff
@@ -15,215 +14,241 @@ import (
 	. "github.com/smartystreets/goconvey/convey"
 )
 
-func Test_FetchState(t *testing.T) {
-	Convey("Verify Fetching State handler", t, func() {
-		var baseTime = time.Now().UTC().Round(time.Second)
-		var testPort = service.Port{Type: "tcp", Port: 8000, ServicePort: 8000}
+var (
+	origSidecarHTTPClient = sidecarHTTPClient
+	origWatcherHTTPClient = watcherHTTPClient
+	dummyState            *catalog.ServicesState
+)
 
-		httpmock.Activate()
-		defer httpmock.DeactivateAndReset()
+func setup() {
+	sidecarHTTPClient = http.DefaultClient
+	watcherHTTPClient = http.DefaultClient
+
+	dummyState = catalog.NewServicesState()
+	dummyState.AddServiceEntry(
+		service.Service{
+			ID:       "007",
+			Name:     "web",
+			Hostname: "some-aws-host",
+			Status:   0,
+			Ports: []service.Port{
+				service.Port{
+					Type:        "tcp",
+					Port:        8000,
+					ServicePort: 8000,
+				},
+			},
+		},
+	)
+
+	httpmock.Activate()
+
+	httpmock.RegisterResponder(http.MethodGet, "http://some.dummy.service",
+		func(req *http.Request) (*http.Response, error) {
+			return httpmock.NewStringResponse(200, ""), nil
+		},
+	)
+
+	httpmock.RegisterResponder(http.MethodGet, "http://some.dummy.service/state.json",
+		func(req *http.Request) (*http.Response, error) {
+			resp, err := httpmock.NewJsonResponse(200, dummyState)
+			if err != nil {
+				return httpmock.NewStringResponse(500, ""), nil
+			}
+			return resp, nil
+		},
+	)
+}
 
-		httpmock.RegisterResponder("GET", "http://some.dummy.service/state.json",
-			func(req *http.Request) (*http.Response, error) {
+func teardown() {
+	httpmock.DeactivateAndReset()
 
-				service := service.Service{ID: "007", Name: "api", Hostname: "some-aws-host",
-					Updated: baseTime, Status: 1, Ports: []service.Port{testPort}}
-				returnState := catalog.NewServicesState()
-				returnState.AddServiceEntry(service)
-				resp, err := httpmock.NewJsonResponse(200, returnState)
-				if err != nil {
-					return httpmock.NewStringResponse(500, ""), nil
-				}
-				return resp, nil
-			},
-		)
+	sidecarHTTPClient = origSidecarHTTPClient
+	watcherHTTPClient = origWatcherHTTPClient
+}
+
+func TestSidecarFetchState(t *testing.T) {
+	setup()
```

Comment:
This is working against the grain for goconvey. Have a read here: https://github.com/smartystreets/goconvey/wiki/Execution-order